### DISCUSSION TITLE
Color Converter

### DISCUSSION THREAD

**Discussion by eduardoagr** [CommunityToolkit/Maui#1496] [New Feature Discussions]
I encountered a compatibility issue with Android in .NET MAUI when working with Maui.Color. In Android, it's not compatible, so I had to find an alternative solution. I decided to store color values as strings and convert them to HEX, which is a string representation.

I initially explored the MauiToolkit to see if there was an existing solution for this problem, but unfortunately, I couldn't find one. As a result, I decided to create my own small converter to address this issue.

While it may seem like a small contribution, I'm genuinely eager to contribute to this amazing community and help make the development experience even better. I appreciate any feedback or assistance in improving this solution further. Thank you for your support!

**Comment by bijington** [comment]
I'm intrigued by the issue. Are you able to share some specifics about the issue and your solution to the problem? I know we have a large number of color related conversion code so if this can improve things that would be great

**Reply by eduardoagr** [reply]
"I'm glad to share my experience.

I was developing a quick NoteApp using .NET MAUI and SQLite. When I attempted to create the property 'Color' in my class, I received the error 'System.NotSupportedException: 'Don't know about Microsoft.Maui.Graphics.Color.''

Also, when saving in SQLite, it needs to be a 'String.'

So, what I did is store the 'Color' as a 'String.'

But what about displaying a hexadecimal value in a label as a .NET MAUI Color? As I recall, a TextColor in a label takes a .NET MAUI Color.

In the Community Toolkit, I found a bunch of converters, but there is no converter to convert the color as a String (hexadecimal) to a .NET MAUI Color.

Doing something like TextColor="{Binding Color}" resulted in `[0:] Microsoft.Maui.Controls.Xaml.Diagnostics.BindingDiagnostics: Warning: '#800080' cannot be converted to type 'Microsoft.Maui.Graphics.Color.''

Apparently, .NET MAUI was trying and failing.

So why not make a Hex to .NET MAUI Color converter?"

**Reply by eduardoagr** [reply]
I know that people don't use SQLite as much these days, but for someone learning .NET MAUI, why not help them out a little?

**Reply by bijington** [reply]
Thank you for explaining this, I understand the issue.

Firstly to cover the part:

> I was developing a quick NoteApp using .NET MAUI and SQLite. When I attempted to create the property 'Color' in my class, I received the error 'System.NotSupportedException: 'Don't know about Microsoft.Maui.Graphics.Color.''

> Also, when saving in SQLite, it needs to be a 'String.'

That is most likely the case. I am going to assume you are using the [sqlite-net-pcl](https://www.nuget.org/packages/sqlite-net-pcl/) package for the Sqlite access. It most likely doesn't provide a mechanism for serialising `Microsoft.Maui.Graphics.Color` so you will need to convert to/from a string.

For the next part:

> In the Community Toolkit, I found a bunch of converters, but there is no converter to convert the color as a String (hexadecimal) to a .NET MAUI Color.

We didn't originally supply this for 2 reasons:

1. There is already the `Color.FromArgb` method that will do this for you.
2. If you supply a string in a XAML binding it will automatically convert that `string` to a `Color` for you. This is done through the use of a `TypeConverter`.

I think our assumption was that point 2 would make it possible for the same behaviour to exist when binding a string value but that does not appear to be the case.

> Doing something like TextColor="{Binding Color}" resulted in `[0:] Microsoft.Maui.Controls.Xaml.Diagnostics.BindingDiagnostics: Warning: '#800080' cannot be converted to type 'Microsoft.Maui.Graphics.Color.''

> Apparently, .NET MAUI was trying and failing.

> So why not make a Hex to .NET MAUI Color converter?"

I am happy with the idea of introducing this and in fact we don't need to introduce new converters, I believe we *just* need to modify our existing converters (e.g. `ColorToRgbStringConverter`, etc.) to be two way converters. Currently they only convert from target to source which only covers the scenario of getting the Color into the data model and not back to the UI.

**Reply by eduardoagr** [reply]
Where can I ubmit the code?

**Reply by bijington** [reply]
Would you be happy to open a PR against this repository?

**Reply by eduardoagr** [reply]
I'll be happy to what is a PR an how to do it

**Reply by bijington** [reply]
That's great thank you.

To start with these links might provide some helpful insight into creating a PR:

https://www.freecodecamp.org/news/how-to-make-your-first-pull-request-on-github-3/

https://www.digitalocean.com/community/tutorials/how-to-create-a-pull-request-on-github

Have a look and let me know if that helps or you need further guidance on those 2 questions. I'll be happy to assist with getting up and running also.

**Reply by eduardoagr** [reply]
I already create a pull request for this

**Comment by eduardoagr** [comment]
I got it


  *   Fork the repository
  *   Clone
  *   Create a branch
  *   Push

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by CliffAgius** [comment]
Closed and answered
